SimpleNotePad.java

```diff
@@ -0,0 +1,213 @@
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setVisible(true);
+		pack();
+	}
+
+	public static void main(String[] args) {
+		SimpleNotePad app = new SimpleNotePad();
+	}
+
+	//extracted methods for each action command
+	@Override
+	public void actionPerformed(ActionEvent e) {
+
+	}
+
+	//new feature: simple-replace method
+	private void action_simpleReplace(ActionEvent e) {
+		String input = JOptionPane.showInputDialog(new JFrame("Input"), "Replace or insert with:");
+		if(input != null) textPane.replaceSelection(input);
+	}
+
+	//new feature: openRecent action method
+	private void action_openRecent(ActionEvent e) {
+		open(new File(e.getActionCommand())); 
+	}
+
+	//open method used by action_openRecent and action_openFile
+	private void open(File file) {
+		updateRecentFiles(file); //updates recent files
+		try {
+			BufferedReader input = new BufferedReader(new InputStreamReader(new FileInputStream(file)));
+			textPane.read(input, "Opening...");    
+		}    			
+		catch (IOException ex) {
+			JOptionPane.showMessageDialog(null,
+					"Open error" + ex, "Opening error",
+					JOptionPane.ERROR_MESSAGE);
+		}
+	}
+
+	//called by open() to update recent files list
+	private void updateRecentFiles(File file) {
+		if(recentFiles.contains(file)) { //removes file from List if it was opened recently to avoid duplicates
+			recentFiles.remove(file);
+		}
+		if(recentFiles.size() == 5) { //removes last item in list if size == 5
+			recentFiles.removeLast();
+		}
+		recentFiles.addFirst(file); //adds file being opened to top of List
+
+		recentMenu.removeAll(); //clears recent menu 
+
+		//repopulates recent menu using updated recent file order
+		for(File f: recentFiles) {
+			JMenuItem recent = new JMenuItem(f.getName());
+			recent.addActionListener(this::action_openRecent);
+			recent.setActionCommand(f.getAbsolutePath());
+			recentMenu.add(recent);
+		} 	
+	}
+
+	//new feature: opens file from system dialog
+	private void action_openFile(ActionEvent e) {
+		JFileChooser fc = new JFileChooser();
+		int returnVal = fc.showOpenDialog(null);
+		if (returnVal == JFileChooser.APPROVE_OPTION) {
+			open(fc.getSelectedFile());
+		}
+	}
+
+	//copy method extracted from actionPerformed
+	private void action_copy(ActionEvent e) {
+		textPane.copy();
+	}
+
+	//paste method extracted from actionPerformed
+	private void action_paste(ActionEvent e) {
+		StyledDocument doc = textPane.getStyledDocument();
+		Position position = doc.getEndPosition();
+		System.out.println("offset"+position.getOffset());
+		textPane.paste();
+	}
+
+	//printFile method extracted from actionPerformed
+	private void action_printFile(ActionEvent e) {
+		try{
+			PrinterJob pjob = PrinterJob.getPrinterJob();
+			pjob.setJobName("Sample Command Pattern");
+			pjob.setCopies(1);
+			pjob.setPrintable(new Printable() {
+				public int print(Graphics pg, PageFormat pf, int pageNum) {
+					if (pageNum>0)
+						return Printable.NO_SUCH_PAGE;
```

Comment:
Good catch to check if the page number when printing the page and checking to see if the page exists.